### PR TITLE
Smoother embark and game screens refactor

### PR DESCRIPTION
**Pull request by joaoburatto**
## Summary

This PR reworks how the game screens work internally, adds a fade out when the player enters the game, creates a ScriptableSettings script to define how general UI works. Reorder objects around the game scene, and creates prefabs from a few objects on the game scene.

### CODE REVIEW

**Review by cosmiccoincidence — DISMISSED** [review]
The following files need to be renamed to PascalCase:
- Player Camer.prefab
- Static World Objects.prefab
- Black Overlay.prefab
- Canvas Template.prefab
- Networked View.prefab

File: Assets/Scripts/SS3D/Systems/Screens/GameScreens.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+using SS3D.Core;
+using SS3D.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace SS3D.Systems.Screens
+{
+    public static class GameScreens
```
Comment (stilnat) [inline Assets/Scripts/SS3D/Systems/Screens/GameScreens.cs:11]: For the sake of clarity, it'd be nice if you could add the [Server] [Client] [ServerOrClient] attributes on relevant methods. That way we understand better what execute what and it limits error due to server or client calling methods they're not supposed to.

File: Assets/Scripts/SS3D/Systems/Screens/GameScreen.cs [hunk 2]
```diff
@@ -12,72 +14,51 @@ namespace SS3D.Systems.Screens
     /// </summary>
     public class GameScreen : Actor
     {
-        [SerializeField] private ScreenType _screenType;
+        public ScreenType ScreenType;
```
Comment (stilnat) [inline Assets/Scripts/SS3D/Systems/Screens/GameScreen.cs]: Should this stay private ?

File: Assets/Scripts/SS3D/Systems/Entities/UI/PlayerCameraBlockerView.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
+using Coimbra.Services.Events;
+using DG.Tweening;
+using SS3D.Core.Behaviours;
+using SS3D.Systems.Entities.Events;
+using UnityEngine;
+
+namespace SS3D.Systems.Entities.UI
```
Comment (stilnat) [inline Assets/Scripts/SS3D/Systems/Entities/UI/PlayerCameraBlockerView.cs:7]: What's this script for, please add comments

**Review by stilnat — CHANGES_REQUESTED** [review]
A few changes to do, seems to work fine with host (I need to mess around more), not tested with client yet. Also need to check if there's conflict with the inventory rework, probably minor ones but please let's do that before any merge, I don't my whole work wasted.

**Comment by stilnat** [comment]
Client breaks the container UI, please check if this happen on develop already, and if not, please fix before merge 
image

**Review by stilnat — APPROVED** [review]
good to go

**Pull request merged**